.github/workflows/go.yml

```diff
@@ -160,19 +160,29 @@ jobs:
           PYROSCOPE_PATH: '${{ steps.pyroscope-path.outputs.PYROSCOPE_PATH }}'
           APPLICATION_NAME: ${{ steps.coverage.outputs.flag }}
 
+      - name: Precompile Tests
+        working-directory: ${{ matrix.package }}
+        run: go test -run=nope ./...
+        env:
+          # no other containers are running at this point so we can use all available memory for compilation
+          GOMEMLIMIT: 5GiB
+          GOGC: -1
+
       - name: Test
         uses: nick-fields/retry@v2
         with:
           command: |
+            set -e
```

Comment:
## Explanation: Why the Script Didn't Stop on Errors

The provided script looped through each package and ran a series of commands on each package. If any command failed during this process, the script would simply continue to the next package in the list without stopping. 

This behavior is due to the default nature of bash scripting. By default, a bash script won't terminate or exit just because a command inside it fails. Instead, it will continue executing subsequent commands.

If you want a bash script to exit immediately when a command fails, you need to add the `set -e` directive at the beginning of the script. The `set -e` directive instructs the shell to exit immediately if any command in the script returns a non-zero exit status (indicating failure).

Without the `set -e` directive, your script will continue running subsequent commands even if one of them fails, which is the behavior you observed.